Server/ConcordiaCurriculumManager/Controllers/DossierReviewController.cs

```diff
@@ -71,4 +76,28 @@ public async Task<ActionResult> ForwardDossier([FromRoute, Required] Guid dossie
 
         return NoContent();
     }
+
+    [HttpPost(nameof(ReviewDossier) + "/{dossierId}")]
+    [Authorize(Policies.IsOwnerOfDossier)]
+    [SwaggerResponse(StatusCodes.Status422UnprocessableEntity, "Invalid input")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    [SwaggerResponse(StatusCodes.Status200OK, "Dossier Review successfully added")]
+    public async Task<ActionResult> ReviewDossier([FromRoute, Required] Guid dossierId, [FromBody, Required] CreateDossierDiscussionMessageDTO dossierMessageDTO)
+    {
+        var dossierMessage = _mapper.Map<DiscussionMessage>(dossierMessageDTO) ?? throw new InvalidInputException("Invalid Message");
+        dossierMessage.DossierDiscussionId = dossierId;
+        await _dossierReviewService.AddDossierDiscussionReview(dossierId, dossierMessage);
+        return NoContent();
+    }
+
+    [HttpPut(nameof(GetAllDossierReviews) + "/{dossierId}")]
+    [SwaggerResponse(StatusCodes.Status422UnprocessableEntity, "Invalid input")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    [SwaggerResponse(StatusCodes.Status200OK, "All Dossier Review were successfully returned")]
+    public async Task<ActionResult> GetAllDossierReviews([FromRoute, Required] Guid dossierId)
```

Comment:
If we create the discussion board when creating the dossier, then remove the DossierDetailsWithDiscussionDTO as I wrote in other comments, then I think we can just call the service layer method to return the dossier details, and the new service layer method `GetDossierWithDiscussion` is not needed. Likewise, this endpoint is not needed either; the user can just query the dossier details.

Reply:
The only reason I created a new endpoint is to reduce the response time. Currently, it takes around 10-20 seconds to retrieve the dossier with discussions (even when I don't have many messages) due to multiple joins. But I'll change it if it's easier for the frontend to just hit a single endpoint.

Reply:
I tested with a dossier with 2 requests, 6 messages, and 2 reviewing groups, and querying the dossier was basically instant on my machine, so I suspect there's some issue on your end (too many programs open simultaneously?)